src/Streamly/Internal/System/Command.hs

```diff
@@ -75,7 +76,12 @@ quotedWord =
                 '"' -> Just x
                 '\'' -> Just x
                 _ -> Nothing
-        trEsc q x = if q == x then Just x else Nothing
+        trEsc q x
+            | q == '\'' && x == '\'' = Just '\''
+            | q == '"' && x == '"' = Just '"'
+            | q == '\'' && x == '\\' = Nothing
+            | q == '"' && x == '\\' = Just '\\'
```

Comment:
Both the double quote cases should be bunched in a single case and then two checks inside that single case.

Reply:
It's alright.